runtime/src/main/java/com/dylibso/chicory/runtime/wasi/WasiP1.java

```diff
@@ -0,0 +1,756 @@
+                        },
+                        "wasi_snapshot_preview1",
+                        "fd_pread",
+                        List.of(I32, I32, I32, I64, I32),
+                        List.of(I32));
+
+        functions[fidx--] =
+                new HostFunction(
+                        (Memory memory, Value... args) -> {
+                            System.out.println("fd_prestat_dir_name: " + Arrays.toString(args));
+                            throw new WASMRuntimeException("We don't yet support this WASI call");
+                            // return new Value[] { Value.i32(0) };
+                        },
+                        "wasi_snapshot_preview1",
+                        "fd_prestat_dir_name",
+                        List.of(I32, I32, I32),
+                        List.of(I32));
+
+        functions[fidx--] =
+                new HostFunction(
+                        (Memory memory, Value... args) -> {
+                            System.out.println("fd_prestat_get: " + Arrays.toString(args));
+                            throw new WASMRuntimeException("We don't yet support this WASI call");
+                            // return new Value[] { Value.i32(0) };
+                        },
+                        "wasi_snapshot_preview1",
+                        "fd_prestat_get",
+                        List.of(I32, I32),
+                        List.of(I32));
+
+        functions[fidx--] =
+                new HostFunction(
+                        (Memory memory, Value... args) -> {
+                            System.out.println("fd_pwrite: " + Arrays.toString(args));
+                            throw new WASMRuntimeException("We don't yet support this WASI call");
+                            // return new Value[] { Value.i32(0) };
+                        },
+                        "wasi_snapshot_preview1",
+                        "fd_pwrite",
+                        List.of(I32, I32, I32, I64, I32),
+                        List.of(I32));
+
+        functions[fidx--] =
+                new HostFunction(
+                        (Memory memory, Value... args) -> {
+                            var fd = args[0].asInt();
+                            InputStream stream;
+                            switch (fd) {
+                                case 0:
+                                    stream = this.options.getStdin();
+                                    break;
+                                default:
+                                    throw new WASMRuntimeException("We don't yet support fd " + fd);
+                            }
+                            if (stream == null) {
+                                throw new WASMRuntimeException(
+                                        "The program attempted an fd_read from"
+                                                + fd
+                                                + "but this was not granted.");
+                            }
```

Comment:
In the future we'll probably keep a map of open files keyed by their fd and do a lookup here. that will require some of the directory mapping stuff which i want to avoid.